src/NServiceBus.RavenDB/Internal/DocumentStoreInitializer.cs

```diff
@@ -53,40 +53,66 @@ void ApplyConventions(ReadOnlySettings settings)
+                    // been run and the TransactionRecoveryStorage cannot be safely swapped with our safe values.
+                    // Only (known) reason to do this would be if customer wants to share the DocumentStore with
+                    // a wider application's persistence needs and must initialize it before NServiceBus is invoked.
+                    // Therefore, we will check to make sure ResourceManagerId and TransactionRecoveryStorage are
+                    // acceptable and throw if they are not.
+
+                    var dtcSettingsAcceptable = store.ResourceManagerId != Guid.Empty
+                                                && store.ResourceManagerId != ravenDefaultResourceManagerId
+                                                && store.TransactionRecoveryStorage is LocalDirectoryTransactionRecoveryStorage;
+
+                    if (!dtcSettingsAcceptable)
+                    {
+                        throw new InvalidOperationException("RavenDB Persistence DocumentStore has already been initialized without safe DTC settings which can lead to data loss. Either allow NServiceBus to initialize the DocumentStore instance or refer to documentation to set up safe DTC settings.");
+                    }
+                }
+                else
+                {
+                    if (store.ResourceManagerId != ravenDefaultResourceManagerId)
+                    {
+                        Logger.Warn("Overriding user-specified documentStore.ResourceManagerId. It's no longer necessary to set this value while using NServiceBus.RavenDB persistence. Consider using busConfiguration.CustomizeDocumentStore(Action<IDocumentStore customize) if this is absolutely necessary, but it is not recommended.");
+                    }
+                    if (!(store.TransactionRecoveryStorage is VolatileOnlyTransactionRecoveryStorage))
+                    {
+                        Logger.Warn("Overriding user-specified documentStore.TransactionRecoveryStorage. It's no longer necessary to set this value while using NServiceBus.RavenDB persistence. Consider using busConfiguration.CustomizeDocumentStore(Action<IDocumentStore customize) if this is absolutely necessary, but it is not recommended.");
+                    }
+
+                    var resourceManagerId = settings.Get<string>("NServiceBus.LocalAddress") + "-" + settings.Get<string>("EndpointVersion");
+                    store.ResourceManagerId = DeterministicGuidBuilder(resourceManagerId);
+
+                    store.TransactionRecoveryStorage = RavenTransactionRecoveryStorageCreator.Create(docStore, settings);
+                }
             }
         }
 
         static Guid DeterministicGuidBuilder(string input)
         {
-            // use MD5 hash to get a 16-byte hash of the string
+            // use SHA1 hash to get a hash of the string - SHA1 to avoid FIPS troubles
```

Comment:
We're also changing the transaction recovery storage so the concern of changing the RMId too is moot. If anything, you want to make sure you have a clean shutdown before you upgrade to this version.